### ISSUE TITLE
Question: Semantics

### ISSUE THREAD

**Issue by nmanginas** [Francesco17/LTLf2DFA#69]
Hi many thanks for the tool. 
I am trying to convert the formula (WX(a) <-> b) for which the automaton is this:

![image](https://github.com/whitemech/LTLf2DFA/assets/138127860/45db7af6-aca0-40b3-8e1c-a725382dc4e7)

Shouldn't the string "~b", i.e. not b be accepted here? Am I missing something?

Thanks

**Comment by francescofuggitti** [comment]
Hi @nmanginas,

thanks for reaching out! Yes, you're right. The string "~b" should be accepted. This is a bug (a nice viz is [here](https://quickstrom.github.io/ltl-visualizer/?formulae=and%28implies%28not%28next%28not%28A%29%29%29%2C+B%29%2C+implies%28B%2C+not%28next%28not%28A%29%29%29%29%29&trace=%7C%7C%7C)). Now, I don't have much time to look into it, but I'll try to do so asap.

Thank you!

**Comment by nickmagginas** [comment]
I am not sure this is wrong after all:
The formula ```WX(a) <-> b``` is equivelant to ```(WX(a) & b) | !(WX a) & ~b)``` which in NNF would correspond to 
```(WX(a) & b) | (X ~a) & ~b)``` per the rules in slide 32 [here](https://ltlf-symposium.github.io/assets/slides/degiacomo.pdf). In that case, I am thinking about this bottom up instead of top down as the compilation done here, it does make sense. Which begs the question what is the formula used to encode the desired behavior which is ```(WX a & b) | (WX ~a) & ~b)```. Which per the rules is not equivalent to ```WX(a) <-> b```. Don't know if this makes sense?

**Comment by nickmagginas** [comment]
Sorry different account :') I am the same person.

**Comment by francescofuggitti** [comment]
The transformation to NNF is correct but also in that case the string "~b" should be accepted. [See the comparison here](https://quickstrom.github.io/ltl-visualizer/?formulae=and%28implies%28not%28next%28not%28A%29%29%29%2C+B%29%2C+implies%28B%2C+not%28next%28not%28A%29%29%29%29%29%7Cor%28and%28next%28not%28A%29%29%2C+not%28B%29%29%2C+and%28B%2C+not%28next%28not%28A%29%29%29%29%29&trace=%7C%7C%7C)

**Comment by JaviMuller** [comment]
I don't think this is a bug. WX(a) is satisfied if there are no more transitions (we have reached the end of the input), or if the next state satisfies a. In the case of the input {~b}, we have that WX(a) is satisfied, as after reading ~b, we have no more input. However, WX(a) is satisfied, in order for the current state to be accepting, b also needs to be satisfied, which is not the case.
